Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1825 @@
+
+```
+ <Loader
+       className="org.apache.catalina.loader.VirtualWebappLoader"
+       virtualClasspath="/usr/local/trippi-sail/*.jar"
+       searchVirtualFirst="true"/>
+```
+Edit `$FEDORA_HOME/server/config/spring/remote-blazegraph.xml` (two edits, in quasi 'diff' format):
+
+In the `<bean id="remoteRepoFactory"` clause insert two lines (careful about the port):
+```
++               <constructor-arg type="java.lang.String" value="http://localhost:8081/blazegraph"/>
++               <constructor-arg type="boolean" value="false"/>
+               <constructor-arg ref="httpClient" />
+               <constructor-arg ref="executorPool" />
+```
+Also modify the `SesameSession` bean (change two lines):
+```
+-        <bean class="org.trippi.impl.sesame.SesameSession">
++        <bean class="org.trippi.impl.sesame.SesameSession" scope="prototype" >
+                <constructor-arg ref="trippiSailRepository" />
+                <constructor-arg ref="org.trippi.AliasManager" />
+-                <constructor-arg value="test://model#" />
++                <constructor-arg value="fedora://model#"/>
+                <constructor-arg value="ri" />
+        </bean>
+```
+
+Edit `$FEDORA_HOME/server/config/fedora.fcfg`:
+
+`cp $FEDORA_HOME/server/config/fedora.fcfg $FEDORA_HOME/server/config/fedora.fcfg.bak`
+
+Comment out the `<param name="datastore" value="localMulgaraTriplestore">` clause.
+
+Edit `$FEDORA_HOME/server/bin/env-server.sh`:
+
+`cp $FEDORA_HOME/server/bin/env-server.sh $FEDORA_HOME/server/bin/env-server.sh.bak`
+
+In the `execWithTheseArgs()` function:
+```
+-	-cp \"$webinf\"/classes:\"$FEDORA_HOME\"/server/bin:\"$webinf\"/lib/* \
++	-cp \"$webinf\"/classes:/usr/local/trippi-sail/*:\"$FEDORA_HOME\"/server/bin:\"$webinf\"/lib/* \
+```
+
+`systemctl stop blazegraph`
+
+Edit `/etc/bigdata/log4j.properties`:
+
+Set the default log level to `ERROR` instead of `WARN` (3 lines near the top).
+
+`systemctl start blazegraph`
+
+`systemctl restart tomcat7`
+
+#### GSearch and Solr <a id="gsearch-solr"></a>
+
+Note: part of the reason we still use older (now very old) versions of Solr is that, as of Solr 5.0 (latest is 7.3), installing Solr as a WAR servlet under containers like Tomcat is no longer supported. Also, quite significant changes in the configuration files have also happened, which makes use of the `basic-solr-config` package from Discovery Garden more difficult. DigiBESS has moved only from Solr 4.2.0 to 4.10.4 and already there are significant changes in `solrconfig.xml` and `schema.xml`. Since some will wish to stick with the tried and true, we include using 4.10.4 as optional. Hopefully this can be part of future migrations to more recent versions of Solr. Set the version of Solr you wish to use in your `install.properties`.
+
+Grab fedoragsearch and solr - do basic config:  
+```
+cd ~  
+
+git clone https://github.com/fcrepo3/gsearch.git
```

Comment:
Should consider using https://github.com/discoverygarden/gsearch which is being maintained by DGI. The current release is 2.9.0.  

Fgs 2.8 does not update the index (it does work on object ingest just not updating) in Tomcat 7.0.73. The issue has arisen due to the following pull request in Tomcat 7.0.73 (similar PR is also in Tomcat8)

https://github.com/apache/tomcat70/commit/cdc0a935c2173aff60039a0b85e57a461381107c

Details can be seen here https://github.com/fcrepo3/gsearch/pull/4

Reply:
Fixed. Thanks! Tested on a deployed machine.